ChatterboxBackend/Chatterbox/Controllers/ChatController.cs

```diff
@@ -0,0 +1,125 @@
+﻿using Chatterbox.Models;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Chatterbox.Controllers
+{
+    [Route("api/[controller]")]
+    public class ChatController:Controller
+    {
+        private readonly ChatterContext _context;
+
+        public ChatController(ChatterContext context)
+        {
+            _context = context;
+        }
+
+        [HttpGet]
+        public IEnumerable<Conversation > GetAll()
+        {
+            return _context.Conversations.ToList();
+        }
+        
+        [HttpPost("newmessage")]
+        public IActionResult Create([FromBody] MessageReceived message)
+        {
+            Console.WriteLine("NEW POST FOR MESSAGE");
+            if (message == null)
+            {
+                Console.WriteLine("ERROR");
+                return BadRequest();
+            }
+            var conversation = _context.Conversations.FirstOrDefault(t => t.ConversationId== message.ConversationId);
+            Message newMessage = new Message();
+            newMessage.DateSent = DateTime.Now;
+            newMessage.UserIdFrom = message.MyId;
+            if (message.MyId != conversation.ConversationRecipientId)
+                newMessage.UserIdto = conversation.ConversationRecipientId;
+            else
+                newMessage.UserIdto = conversation.ConversationStarterId;
+
+            newMessage.Text = message.Message;
+            if (conversation.Messages == null)
+            {
+                conversation.Messages = new List<Message>();
+                Console.WriteLine("NEW MESSAGE");
+            }
+
+            conversation.Messages.Add(newMessage);
+            conversation.LastMessageText = message.Message;
+            conversation.LastMessageDate = newMessage.DateSent;
+
+            //_context.Messages.Add(newMessage);
+            _context.SaveChanges();
+
+            return new NoContentResult(); 
+        }
+
+        [HttpGet("conversations/{id}", Name = "GetTodo")]
+        public IActionResult GetById(long id)
+        {
+            var items = _context.Conversations.Where(c=>(c.ConversationStarterId == id || c.ConversationRecipientId == id));//.ToList();
+
+
+            if (items == null)

```

Comment:
I don't think this will actually work. `Where()` will return an IQueryable, and will never be null. It doesn't actually hit the database.

If you want to return `NotFound` if nothing exists, then query the database by calling `ToList()` and checking `Count()` on the result